src/documentation/dsls/sql/shell.md

```diff
@@ -0,0 +1,176 @@
+
+No rows affected (0.042 seconds)
+```
+
+
+The `CREATE TABLE` statement registers the CSV file as a table in Beam SQL and specifies the table's schema. This statement does not directly create a persistent physical table; it only describes the source/sink to Beam SQL so that you can use the table in the queries that read data and write data.
+
+_For more information about `CREATE TABLE` syntax and supported table types, see the [CREATE TABLE reference page](https://beam.apache.org/documentation/dsls/sql/create-table/)._
+
+
+## Reading and Writing Data
+
+To read data from the local CSV file that you declared in the previous section, execute the following query:
+
+
+```
+0: BeamSQL> SELECT field1 AS field FROM csv_file;
++--------+
+| field  |
++--------+
+| baz    |
+| foo    |
+| bar    |
+| bar    |
+| foo    |
++--------+
+```
+
+
+_For more information about `SELECT` syntax, see the [SELECT reference page](https://beam.apache.org/documentation/dsls/sql/select/)._
+
+To write data to the CSV file, use the `INSERT INTO … SELECT ...` statement:
+
+
+```
+0: BeamSQL> INSERT INTO csv_file SELECT 'foo', 'bar';
+```
+
+Read and write behavior depends on the type of the table. For example:
+
+*   The table type `text` is implemented using `TextIO`, so writing to a `text` table can produce multiple numbered files. 
+*   The table type `pubsub` is an unbounded source, so reading from a `pubsub` table never completes.
+
+## Developing with unbounded Sources
+
+When you want to inspect the data from an unbounded source during development, you must specify the `LIMIT x` clause at the end of the `SELECT` statement to limit the output to `x` number of records. Otherwise, the pipeline will never finish.
+
+```
+0: BeamSQL> SELECT field1 FROM unbounded_source LIMIT 10 ;
+```
+
+The example queries shown so far are fast queries that execute locally. These queries are helpful when you are investigating the data and iteratively designing the pipeline. Ideally, you want the queries to finish quickly and return output when complete. 
+
+When you're satisfied with the logic of your SQL statements, you can submit the statements as long-running jobs by dropping the `LIMIT x` statement. Then, the pipeline can potentially run forever if one of the tables represents an unbounded source.
+
+
+## Specifying the Runner
+
+By default, Beam uses the `DirectRunner` to run the pipeline on the machine where you're executing the commands. If you want to run the pipeline with a different runner, you must perform two steps:
+
+1.  Make sure the SQL shell includes the desired runner. Add the corresponding project id to the `-Pbeam.sql.shell.bundled` parameter of the Gradle invocation ([source code](https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/shell/build.gradle#L26), [project ids](https://github.com/apache/beam/blob/master/settings.gradle)). For example, use the following command to include Flink runner:
```

Comment:
Same comment about line numbers in master.

Reply:
Dropped the line number